drivers/input/input_vs1838b.c

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2025 Bastien Jauny <bastien.jauny@smile.fr>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT vishay_vs1838b
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(input_vs1838b, CONFIG_INPUT_LOG_LEVEL);
+
+/* A NEC packet is defined by:
+ * - a lead burst (2 edges)
+ * - an 8-bit address followed by its logical inverse
+ * - an 8-bit command followed by its logical inverse
+ * - a trailing burst
+ */
+
+/* Constants used for parsing the edges buffer for NEC protocol */
+#define NEC_LEAD_PULSE_EDGE_OFFSET 0
+#define NEC_LEAD_PULSE_EDGE_WIDTH  2
+
+#define NEC_ADDRESS_BYTE_EDGE_OFFSET ((NEC_LEAD_PULSE_EDGE_OFFSET) + (NEC_LEAD_PULSE_EDGE_WIDTH))
```

Comment:
```suggestion
#define NEC_ADDRESS_BYTE_EDGE_OFFSET (NEC_LEAD_PULSE_EDGE_OFFSET + NEC_LEAD_PULSE_EDGE_WIDTH)
```

the convention is that you'd expect the macro to have outer parenthesis in the definition if needed, or wrap variables, no need to add them both on definition and usage

same on the others below